source/frontend/src/features/properties/map/agreement/update/AgreementSubForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { FastCurrencyInput, FastDatePicker, Input, Select, TextArea } from 'components/common/form';
+import { YesNoSelect } from 'components/common/form/YesNoSelect';
+import { SectionField } from 'features/mapSideBar/tabs/SectionField';
+import { FormikProps } from 'formik';
+import React from 'react';
+import { ILookupCode } from 'store/slices/lookupCodes';
+import styled from 'styled-components';
+import { mapLookupCode } from 'utils';
+import { withNameSpace } from 'utils/formUtils';
+
+import { StyledSectionSubheader } from '../styles';
+import { AgreementsFormModel } from './models';
+
+export interface IAgreementSubFormProps {
+  index: number;
+  nameSpace: string;
+  formikProps: FormikProps<AgreementsFormModel>;
+  agreementTypes: ILookupCode[];
+}
+
+export const AgreementSubForm: React.FunctionComponent<IAgreementSubFormProps> = ({
+  index,
+  nameSpace,
+  formikProps,
+  agreementTypes,
+}) => {
+  const H0074Type = 'H0074';
+  return (
+    <>
+      <StyledSectionSubheader>Agreement details</StyledSectionSubheader>
+      <SectionField labelWidth="5" label="Agreement status">
+        <YesNoSelect field={withNameSpace(nameSpace, 'agreementStatus')} notNullable />
+      </SectionField>
+      <SectionField labelWidth="5" label="Legal survey plan">
+        <Input field={withNameSpace(nameSpace, 'legalSurveyPlanNum')} />
+      </SectionField>
+      <SectionField labelWidth="5" label="Agreement type" required>
+        <Select
+          field={withNameSpace(nameSpace, 'agreementTypeCode')}
+          options={agreementTypes.map(mapLookupCode)}
+          placeholder="Select Agreement Type"
+          onChange={(e: React.ChangeEvent<HTMLSelectElement>) => {
+            const selectedValue = [].slice
+              .call(e.target.selectedOptions)
+              .map((option: HTMLOptionElement & number) => option.value)[index];
+            if (!!selectedValue && selectedValue !== H0074Type) {
+              formikProps.setFieldValue(withNameSpace(nameSpace, 'commencementDate'), '');
```

Comment:
hmm, perhaps this isn't working then? didn't seem to be resetting the field in my tests.

Reply:
also seems like a good unit test candidate.

Reply:
odd, it works when I tried it